lvgl/src/lv_core/style.rs

```diff
@@ -64,1144 +64,197 @@ impl From<Opacity> for u8 {
-                value,
-            );
-        }
-    }
-
-    pub fn set_border_blend_mode(&mut self, state: State, value: i16) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_int(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_BORDER_BLEND_MODE as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value,
-            );
-        }
-    }
-
-    pub fn set_border_post(&mut self, state: State, value: i16) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_int(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_BORDER_POST as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value,
-            );
-        }
-    }
-
-    pub fn set_border_color(&mut self, state: State, value: Color) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_color(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_BORDER_COLOR as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value.raw,
-            );
-        }
-    }
-
-    pub fn set_border_opa(&mut self, state: State, value: Opacity) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_opa(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_BORDER_OPA as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value.into(),
-            );
-        }
-    }
-
-    pub fn set_outline_width(&mut self, state: State, value: i16) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_int(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_OUTLINE_WIDTH as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value,
-            );
+macro_rules! gen_lv_style {
+    ($func_name:ident,$lv_name:ident,$vty:ty) => {
+        #[inline]
+        pub fn $func_name(&mut self, value: $vty) {
+            unsafe {
+                lvgl_sys::$lv_name(
+                    self.raw.as_mut(),
+                    value.into(),
+                );
+            }
         }
     }
+}
 
-    pub fn set_outline_pad(&mut self, state: State, value: i16) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_int(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_OUTLINE_PAD as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value,
-            );
+macro_rules! gen_lv_style_color {
```

Comment:
Not super happy with the existence of this macro; I'll probably implement `Into<lv_color_t> for Color` so I can reuse the other macro

Reply:
Fixed!